src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/AnalyzerTest`1.cs

```diff
@@ -1265,6 +1272,47 @@ static Solution AddProjectReferences(Solution solution, ProjectId sourceProject,
             }
         }
 
+        protected (string fileName, IEnumerable<string> folders) GetNameAndFoldersFromPath(string projectPathPrefix, string path)
+        {
+            // Normalize to platform path separators for simplicity later on
+            var normalizedPath = path.Replace(Path.AltDirectorySeparatorChar, Path.DirectorySeparatorChar);
+            var normalizedDefaultPathPrefix = projectPathPrefix.Replace(Path.AltDirectorySeparatorChar, Path.DirectorySeparatorChar);
+
+            if (!Path.IsPathRooted(normalizedDefaultPathPrefix))
+            {
+                // If our default path isn't rooted, then we assume that we don't have any rooted paths
+                // and just use the file name
+                return (Path.GetFileName(normalizedPath), NoFoldersArray);
+            }
+
+            var projectRootPath = Path.GetFileName(normalizedDefaultPathPrefix) == string.Empty
```

Comment:
💡 A comment on this would be good.

| Default path | Project root path |
| --- | --- |
| `/0/Temp` | `/0/` |
| `/0/` | `/0/` |